fix(bitget) - ws types reorg 1

ttodua opened the pull request:
works with (tested and the affected market orders are correct):
```
await e.watchOrders (undefined, undefined, undefined, {subType:'linear', settle:'USDT'});
await e.watchOrders (undefined, undefined, undefined, {subType:'linear', settle:'USDC'});
await e.watchOrders (undefined, undefined, undefined, {subType:'linear'});  // defaults to USDT
await e.watchOrders (undefined, undefined, undefined, {subType:'inverse'});
await e.watchOrders (undefined, undefined, undefined, {type:'swap'});  // defaults to LINEAR, USDT


await e.watchOrders (undefined, undefined, undefined, {productType:'USDT-FUTURES'});
await e.watchOrders (undefined, undefined, undefined, {productType:'USDC-FUTURES'});
await e.watchOrders (undefined, undefined, undefined, {productType:'COIN-FUTURES'});
```

the method (watchOrders) implementation is also simplified imo, and what's more, the added helper method can be reused in other implementations

ttodua committed:
fix(bitget) - ws types reorg 1

ttodua committed:
fix(bitget) - multiple handlings

ttodua committed:
minor

ttodua committed:
order handling

ttodua committed:
lint

ttodua committed:
lint2

ttodua committed:
minor

ttodua committed:
back rev

ttodua committed:
move to Rest

carlosmiei commented:
@ttodua unfortunately I'm merging a bug fix that will clash with this one I think

carlosmiei commented:
this pr https://github.com/ccxt/ccxt/pull/22734

ttodua committed:
-
Merge branch 'master' of https://github.com/ccxt/ccxt into bitget-watch-orders-2

# Conflicts:
#	ts/src/pro/bitget.ts

ttodua commented:
@carlosmiei i've fixed conflicts

carlosmiei commented on ts/src/pro/bitget.ts:
```diff
@@ -993,43 +993,33 @@ export default class bitget extends bitgetRest {
-        let subType = undefined;
-        [ subType, params ] = this.handleSubTypeAndParams ('watchOrders', market, params, 'linear');
-        if ((type === 'spot' || type === 'margin') && (symbol === undefined)) {
+        [ params, type, subType, settle, instType ] = this.handleProductTypesAndParams (market, 'watchOrders', params);
```
This method is very very confusing, and the breaks the pattern of the handleXAndParams that usually returns the `[option, params]` and not a bunch of options
